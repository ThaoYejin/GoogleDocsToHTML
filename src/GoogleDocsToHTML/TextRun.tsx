
import { docs_v1 } from "googleapis";
import React from "react";


export interface TextRunProps {
    paragraph?: docs_v1.Schema$Paragraph;
}

export const TextRun: React.FC<TextRunProps> = ({
    paragraph
}) => {
    var elements = paragraph?.elements;
    var content: (string | undefined)[];
    content = [];
    elements?.forEach(function (value) {
        content.push(value?.textRun?.content);
    })
    var contentReturn = "";
    content.forEach(function (value) {
        contentReturn += value;
    })
    return (
        <>
            {contentReturn}
        </>
    );
}