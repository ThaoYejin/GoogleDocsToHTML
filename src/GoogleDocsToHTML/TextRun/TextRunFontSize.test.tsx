import React from 'react';
import renderer from 'react-test-renderer';
import { TextRun } from './TextRun';

it('renders correctly', () => {
    var element = {
        "startIndex": 1,
        "endIndex": 5,
        "textRun": {
          "content": "Text",
          "textStyle": {
            "foregroundColor": {
              "color": {
                "rgbColor": {
                  "red": 0.2509804,
                  "green": 0.2509804,
                  "blue": 0.2509804
                }
              }
            },
            "fontSize": {
              "magnitude": 45,
              "unit": "PT"
            }
          }
        }
      };

    const tree = renderer
        .create(<TextRun element={element} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})