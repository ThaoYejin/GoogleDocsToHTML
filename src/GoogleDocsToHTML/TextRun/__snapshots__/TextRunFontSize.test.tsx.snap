// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  style={
    Object {
      "color": "rgb(64.000002,64.000002,64.000002)",
      "fontSize": 45,
    }
  }
>
  Text
</span>
`;
